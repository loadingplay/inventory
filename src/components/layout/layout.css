@import url('https://fonts.googleapis.com/css?family=Montserrat:400,700&display=swap');
html,
body {
  height: 100%;
  margin: 0px;
}
body {
  background: #e9e9e9;
  color: #666666;
  font-family: 'Montserrat', sans-serif;
  font-size: 15px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.login_wrapper {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 100%;
  justify-items: center;
  align-items: center;
  height: 80vh;
}

/* Form Module */
.form-module {
  position: relative;
  background: #ffffff;
  max-width: 320px;
  width: 100%;
  border-top: 5px solid #33b5e5;
  box-shadow: 0 0 3px rgba(0, 0, 0, 0.1);
  margin: 0 auto;
}
.form-module .form {
  padding: 40px;
}
.logo {
  text-align: center;
}
.form-module h2 {
  margin: 20px 0;
  color: #33b5e5;
  font-size: 18px;
  font-weight: 400;
  line-height: 1;
}
.form-module input {
  outline: none;
  display: block;
  width: 100%;
  border: 1px solid #d9d9d9;
  margin: 0 0 20px;
  padding: 10px 15px;
  box-sizing: border-box;
  font-weight: 400;
  transition: 0.3s ease;
}
.form-module input:focus {
  border: 1px solid #33b5e5;
  color: #333333;
}
.form-module button {
  cursor: pointer;
  background: #33b5e5;
  width: 100%;
  border: 0;
  padding: 10px 15px;
  color: #ffffff;
  transition: 0.3s ease;
}
.form-module button:hover {
  background: #178ab4;
}
.form-module .cta {
  background: #f2f2f2;
  width: 100%;
  padding: 15px 40px;
  box-sizing: border-box;
  color: #666666;
  font-size: 12px;
  text-align: center;
}
.form-module .cta a {
  color: #333333;
  text-decoration: none;
}

/*google_login section*/

.google_login button {
  cursor: pointer;
  background: #dd4b39;
  width: 100%;
  border: 0;
  padding: 10px 15px;
  color: #ffffff;
  transition: 0.3s ease;
}
.google_login button:hover {
  background: #b52f1f;
}

.google_login,  button::-moz-focus-inner {
  border: 0;
}
