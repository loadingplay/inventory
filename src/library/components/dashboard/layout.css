body {
    padding-top: 4.5rem;
}

.toast-container {
    width: 100%;
    position: absolute;
    bottom: 150px;
    padding: 10px;
}

.toast-content {
    background: #333;
    color: #ccc;
    border-radius: 5px;
    padding: 10px;
    width: fit-content;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    font-size: 25px;
}
