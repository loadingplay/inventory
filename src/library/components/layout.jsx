import React, { PureComponent } from 'react'
import PropTypes from "prop-types";

export default class Layout extends PureComponent {

  propTypes = {
    children: PropTypes.object
  }

  render() {
    let { children } = this.props
    return (
      <div className="login_wrapper">
        {children}
      </div>
    )
  }
}
