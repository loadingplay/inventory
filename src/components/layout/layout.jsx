import React, { Component } from 'react'
import './layout.css'

export default class layout extends Component {
  render() {
    let { children } = this.props
    return (
      <div className="login_wrapper">
        {children}
      </div>
    )
  }
}
