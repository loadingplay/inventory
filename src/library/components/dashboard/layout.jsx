import React, { Component } from 'react';
import PropTypes from "prop-types";

import { LogoutButton } from '../../components';

export class DashboardLayout extends Component {

  static propTypes = {
    title: PropTypes.string,

    menuTitle: PropTypes.string,
    menu: PropTypes.object,

    tableTitle: PropTypes.string,
    table: PropTypes.object,
    tableControls: PropTypes.object
  }

  render = () => {
    return (
      <div className="container dashboard_wrapper">
        <div className="dashboard_elements row">
          <h3 className="dashboard_title col-12">
            {this.props.title}
          </h3>

          {/* Stores */}
          <div className="col-2" >
            <div className="subtitle">{this.props.menuTitle}</div>
            {this.props.menu}
          </div>

          {/* Inventory table */}
          <div className="col-10 row" >
            <div className="subtitle col-6">
              {this.props.tableTitle}
            </div>
            <div className="col-6">
              <LogoutButton></LogoutButton>
            </div>
            {this.props.table}

            {/* Controls */}
            <section className="controls col-12 row">
              {this.props.tableControls}
            </section>
          </div>
        </div>
      </div>
    )
  }
}
