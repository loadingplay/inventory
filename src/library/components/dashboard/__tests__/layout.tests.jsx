import React from 'react';
import { shallow } from "enzyme";

import { DashboardLayout } from "../layout";

describe("DashboardLayout", () => {
  test("it should render properly", () => {
    const wrapper = shallow(<DashboardLayout />);
    expect(wrapper).toMatchSnapshot("detault");
  });
});
