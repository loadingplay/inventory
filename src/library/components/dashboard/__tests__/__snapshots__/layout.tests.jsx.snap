// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardLayout it should render properly: detault 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <h3
      className="dashboard_title col-12"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-2"
    >
      <div
        className="subtitle"
      />
    </div>
    <div
      className="col-10"
    >
      <div
        className="row col-12"
      >
        <div
          className="subtitle col-6"
        />
        <div
          className="col-6"
        >
          <LogoutButton />
        </div>
      </div>
      <section
        className="controls col-12 row"
      />
    </div>
  </div>
</div>
`;
